src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpGoToImplementation.cs

```diff
@@ -92,5 +91,55 @@ public void SomeMethod()
             VisualStudio.Editor.GoToImplementation("FileImplementation.cs");
             VisualStudio.Editor.Verify.TextContains(@"class Implementation$$ : IDisposable", assertCaretPosition: true);
         }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.GoToImplementation)]
+        public void GoToImplementationFromSourceAndMetadata()
+        {
+            var project = new ProjectUtils.Project(ProjectName);
+            VisualStudio.SolutionExplorer.AddFile(project, "FileImplementation.cs");
+            VisualStudio.SolutionExplorer.OpenFile(project, "FileImplementation.cs");
+            VisualStudio.Editor.SetText(
+@"using System;
+
+class Implementation : IDisposable
+{
+    public void Dispose()
+    {
+    }
+}");
+            VisualStudio.SolutionExplorer.CloseCodeFile(project, "FileImplementation.cs", saveFile: true);
+
+            VisualStudio.SolutionExplorer.AddFile(project, "FileUsage.cs");
+            VisualStudio.SolutionExplorer.OpenFile(project, "FileUsage.cs");
+            VisualStudio.Editor.SetText(
+@"using System;
+
+class C
+{
+    void M()
+    {
+        IDisposable c;
+        try
+        {
+            c = new Implementation();
+        }
+        finally
+        {
+            c.Dispose();
+        }
+    }
+}");
+
+            VisualStudio.Editor.PlaceCaret("Dispose", charsOffset: -1);
+
+            // It should not navigate to the implementation so we stay on the same file
+            VisualStudio.Editor.GoToImplementation("FileUsage.cs");
+
+            var results = VisualStudio.FindReferencesWindow.GetContents("'Dispose' implementations");
+
+            // There are a lot of results, no point transcribing them all into a test
```

Comment:
we had taht with a test back in the native IDE days.  and every time the bcl would change our test would break :D